extraConfigDpu.py

```diff
@@ -311,12 +320,12 @@ def ExtraConfigDpu(cc: ClustersConfig, cfg: ExtraConfigArgs, futures: dict[str,
 
 def ExtraConfigDpuHost(cc: ClustersConfig, cfg: ExtraConfigArgs, futures: dict[str, Future[Optional[host.Result]]]) -> None:
     logger.info("Running post config step to start DPU operator on Host")
-
     lh = host.LocalHost()
     client = K8sClient(cc.kubeconfig)
+    pullnumber = cc.dpu_operator_pull_number
```

Comment:
More generally, why does CDA need to know about the concept of a pull number?

In my mind, I assumed whatever CI script calls CDA would be responsible for taking the poll number as an argument, and would hand the corresponding repo and branch to CDA as a config argument as in https://github.com/bn222/cluster-deployment-automation/pull/217.

It feels a bit like the logic for the CI is bleeding into CDA here. CDA is just the tool to deploy in my mind, CDA is just the deployment tool used by the CI, not the CI itself.

But perhaps I am thinking about this wrong.

Reply:
We do need a way to build a not-yet-merged PR. Otherwise I'm on board with your comment. The better way to avoid that is to allow to pass in a path which could be `..` if cda is a submodule in dpu-operator.

Reply:
I don't think there is a significant difference between the two options. However, if CDA needs a pull number to be converted to a branch number, it is probably best to do it inside CDA so we don't "outsource" functionality to a different script. Including the conversion functionality within the CDA makes the codebase more self-contained and reduces dependencies on external scripts. Overall, it doesn't seem to matter that much.

Reply:
> We do need a way to build a not-yet-merged PR

So PR with poll number 123 comes in to DPU operator, can't I just use the github api to see this is PR is coming from repo "https://github.com/vrindle/dpu-operator.git" branch "test_branch" and then run cda with the post config:
```
    - name: "dpu_operator_dpu"
      dpu_operator_repo:
"https://github.com/vrindle/dpu-operator.git"
      dpu_operator_branch: "test_branch"
```

To me this is a bit nicer, keeps the scope of CDA cleaner

Reply:
For sure CDA won't be running the E2E tests for the DPU Operator. It will probably reside in a different set of scripts. Which means it probably needs to check that the DPU Operator git repo is checked out. So the logic for pull number may be more appropriate to be there instead.

Reply:
Nice as a dev too, because I don't need to submit a PR to deploy my repo/branch, just push my changes and have a config file pointing here.

Reply:
okay That's fine. I will address these comments.

Reply:
removed pull number